packages/flutter/test/widgets/selectable_text_test.dart

```diff
@@ -4252,4 +4376,171 @@ void main() {
+      ),
+    );
+
+    expect(selection, isNull);
+
+    // Put the cursor at the end of the field.
+    await tester.tapAt(textOffsetToPosition(tester, 10));
+    expect(selection, isNotNull);
+    expect(selection!.baseOffset, 10);
+    expect(selection!.extentOffset, 10);
+
+    // Double tapping the second space selects it.
+    await tester.pump(const Duration(milliseconds: 500));
+    await tester.tapAt(textOffsetToPosition(tester, 5));
+    await tester.pump(const Duration(milliseconds: 50));
+    await tester.tapAt(textOffsetToPosition(tester, 5));
+    await tester.pumpAndSettle();
+    expect(selection, isNotNull);
+    expect(selection!.baseOffset, 5);
+    expect(selection!.extentOffset, 6);
+
+    // Put the cursor at the end of the field.
+    await tester.tapAt(textOffsetToPosition(tester, 10));
+    expect(selection, isNotNull);
+    expect(selection!.baseOffset, 10);
+    expect(selection!.extentOffset, 10);
+
+    // Double tapping the second space selects it.
+    await tester.pump(const Duration(milliseconds: 500));
+    await tester.tapAt(textOffsetToPosition(tester, 0));
+    await tester.pump(const Duration(milliseconds: 50));
+    await tester.tapAt(textOffsetToPosition(tester, 0));
+    await tester.pumpAndSettle();
+    expect(selection, isNotNull);
+    expect(selection!.baseOffset, 0);
+    expect(selection!.extentOffset, 1);
+  }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.macOS,  TargetPlatform.windows, TargetPlatform.linux, TargetPlatform.fuchsia }));
+
+  testWidgets('double tapping a space selects the previous word on mobile', (WidgetTester tester) async {
+    TextSelection? selection;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: Center(
+            child: SelectableText(
+              ' blah blah  \n  blah',
+              onSelectionChanged: (TextSelection newSelection, SelectionChangedCause? cause){
+                selection = newSelection;
+              },
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(selection, isNull);
+
+    // Put the cursor at the end of the field.
+    await tester.tapAt(textOffsetToPosition(tester, 19));
+    expect(selection, isNotNull);
+    expect(selection!.baseOffset, 19);
+    expect(selection!.extentOffset, 19);
+
+    // Double tapping does the same thing
+    await tester.pump(const Duration(milliseconds: 500));
+    await tester.tapAt(textOffsetToPosition(tester, 5));
+    await tester.pump(const Duration(milliseconds: 50));
+    await tester.tapAt(textOffsetToPosition(tester, 5));
+    await tester.pumpAndSettle();
+    expect(selection, isNotNull);
+    expect(selection!.baseOffset, 1);
+    expect(selection!.extentOffset, 5);
+
+    // Put the cursor at the end of the field.
+    // await tester.tapAt(textOffsetToPosition(tester, 19));
+    // expect(selection, isNotNull);
+    // expect(selection!.baseOffset, 19);
+    // expect(selection!.extentOffset, 19);
```

Comment:
Does moving the cursor to the end of the field first break something?

Reply:
Yes, for some reason I am unable to put the cursor at the end of the field at this point in the test. I tested a couple of offset positions and came to the conclusion that anything past the value `17` results in the cursor staying at the same previous position, which in this case is `1`. However, just a little later in the same test, I am able to move the cursor to the end of the field. I haven't been able to pinpoint yet what exactly is causing this.

Reply:
So I tried a similar test on master

```dart
  testWidgets('double tapping a space selects the space on mobile', (WidgetTester tester) async {
    TextSelection? selection;

    await tester.pumpWidget(
      MaterialApp(
        home: Material(
          child: Center(
            child: SelectableText(
              ' blah blah  \n  blah',
              onSelectionChanged: (TextSelection newSelection, SelectionChangedCause? cause){
                selection = newSelection;
              },
            ),
          ),
        ),
      ),
    );

    expect(selection, isNull);

    // Put the cursor at the end of the field.
    await tester.tapAt(textOffsetToPosition(tester, 19));
    expect(selection, isNotNull);
    expect(selection!.baseOffset, 19);
    expect(selection!.extentOffset, 19);

    // Double tapping the second space selects the space.
    await tester.pump(const Duration(milliseconds: 500));
    await tester.tapAt(textOffsetToPosition(tester, 5));
    await tester.pump(const Duration(milliseconds: 50));
    await tester.tapAt(textOffsetToPosition(tester, 5));
    await tester.pumpAndSettle();
    expect(selection, isNotNull);
    expect(selection!.baseOffset, 5);
    expect(selection!.extentOffset, 6);

    // Put the cursor at the end of the field.
    await tester.tapAt(textOffsetToPosition(tester, 19));
    expect(selection, isNotNull);
    expect(selection!.baseOffset, 19);
    expect(selection!.extentOffset, 19);
    
  }, variant: const TargetPlatformVariant(<TargetPlatform>{ TargetPlatform.iOS, TargetPlatform.android }));
```
The test also fails when trying to do the same behavior (put the cursor at the end of the field)

Reply:
Sounds good, thanks for checking.  Remove these commented out lines then?